homework-submission/week3_hw/react-hyf/client/src/App.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import './App.css';
+import Button from './components/button';
+import TodoItems from './components/createDynamicList';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [],
+      newTodo: '',
+    };
+  }
+  componentDidMount() {
+    //fetch todos
+    this.fetchTodos().then(todos => {
+      this.setState({
+        todos,
+      });
+    });
+  }
+
+  fetchTodos = async () => {
+    const response = await fetch('/todos');
+    const parsedJSON = await response.json();
+    return parsedJSON;
+  };
+
+  //Delete Todos
+  deleteAllTodos = async () => {
+    const response = fetch('/todos', {
+      method: 'DELETE',
+    });
+    response.then(data => console.log(data));
+    this.setState({
+      todos: [],
+    });
+  };
+
+  // Delete single todo
+  deleteTodo = key => {
+    const response = fetch('/todos/:id', {
+      method: 'Delete',
+    });
+    const todos = [...this.state.todos];
+    const filteredItems = [...this.state.todos].filter(item => {
+      return item.key !== key;
+    });
+    response.then(data => console.log(data));
+
+    this.setState({
+      todos: filteredItems,
+    });
+  };
+
+  SubmitAddTodo = event => {
+    const response = fetch('/todos', {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json, text/plain',
+        'Content-type': 'application/json',
+      },
+      body: JSON.stringify({ key: Date.now(), name: this.state.newTodo }),
```

Comment:
`{ key: Date.now(), name: this.state.newTodo }` is an object used twice with a different date time stamp.
Making it generates two different timings
 - one for the server, when the request happend
 - and other for the client when the response received

you can set these into a variable so it would make it easier to read, and debug.